lib/cambiatus/orders.ex

```diff
@@ -77,9 +78,11 @@ defmodule Cambiatus.Orders do
   """
 
   # TODO: Add current_user to query
-  def get_shopping_cart() do
+  def get_shopping_cart(buyer) do
     Order
     |> where([o], o.status == "cart")
+    |> join(:left, [o], b in assoc(o, :buyer))
+    |> where([o, b], b.account == ^buyer.account)
```

Comment:
can't we simply preload the buyer? Something like this:

```suggestion
    |> Repo.preload(:buyer)
```

Reply:
I don't think so. Because this is getting an order where the status is `cart` and belongs to the current user. Since we're getting the order from these SQL statements I don't believe it's possible to preload the buyer.